src/gui/dialogs/file_dialog.cpp

```diff
@@ -196,12 +196,20 @@ void file_dialog::check_filename() {
 			}
 		}
 
+		utils::string_map i18n_strings;
+		i18n_strings["red_x_mark"] = "<span color='red' face='DejaVuSans'>✘</span>";
 		if (wrong_ext) {
-			validation_msg.set_label("<span color='red' face='DejaVuSans'>✘</span><span color='red' size='small'>wrong extension, use " + utils::join(extensions_, ", ") + "</span>");
+			i18n_strings["extensions"] = utils::join(extensions_, ", ");
+			validation_msg.set_label(VGETTEXT("$red_x_mark <span color='red' size='small'>wrong extension, use $extensions</span>", i18n_strings));
 			save_btn.set_active(false);
 		} else {
-			validation_msg.set_label("");
-			save_btn.set_active(true);
+			if (filename.find(' ') != std::string::npos) {
+				validation_msg.set_label(VGETTEXT("$red_x_mark <span color='red' size='small'>whitespace is not allowed</span>", i18n_strings));
```

Comment:
What about just keeping the label static in the CFG and hiding/showing it as needed instead of setting the label here?